cli/command/container/stats.go

```diff
@@ -264,31 +264,40 @@ func RunStats(ctx context.Context, dockerCLI command.Cli, options *StatsOptions)
 		// so we unlikely hit this code in practice.
 		daemonOSType = dockerCLI.ServerInfo().OSType
 	}
+
+	// Buffer to store formatted stats text.
+	// Once formatted, it will be printed in one write to avoid screen flickering.
+	var statsTextBuffer strings.Builder
```

Comment:
I'd suggest using `bytes.Buffer` here instead, see my comment below
```suggestion
	var statsTextBuffer bytes.Buffer
```